---
title: "Tau Mutant GeoMx Analysis"
author:
  - name: "Emir Turkes and Naoto Watamura, UK DRI at UCL"
date: '`r strftime(Sys.time(), format = "%B %d, %Y")`'
link-citations: true
output:
  html_document:
    code_folding: show
    number_sections: true
    theme: lumen
    highlight: haddock
    toc: true
    toc_depth: 2
    toc_float:
      collapsed: false
      smooth_scroll: false
knit: (function(inputFile, encoding) {
  rmarkdown::render(
    inputFile,
    encoding = encoding,
    output_file = file.path("..", "results", "tau_mutant_geomx_analysis.html")
  )})
---

<style type="text/css">
body {font-size: 16px;}
h1.title {font-size: 35px;}
h1 {font-size: 24px;}
h2 {font-size: 22px;}
.toc-content {padding-left: 0px; padding-right: 0px;}
div.tocify {width: 100%;}
.tocify-subheader .tocify-item {
  font-size: 0.95em; padding-left: 25px; text-indent: 0;
}
div.main-container {max-width: none; width: 100%;}
</style>

```{r}
#    This file is part of tau-mutant-geomx.
#    Copyright (C) 2025  Emir Turkes, Naoto Watamura, UK DRI at UCL
#
#    This program is free software: you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation, either version 3 of the License, or
#    (at your option) any later version.
#
#    This program is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License
#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
#    Emir Turkes can be contacted at emir.turkes@eturkes.com

# Load packages.
# --------------
library(conflicted)
packages <- c("GeomxTools", "Seurat")
invisible(suppressPackageStartupMessages(lapply(packages, FUN = library, character.only = TRUE)))
# --------------

# Add paths.
# ----------
data_dir <- file.path("..", "data")
cache_dir <- file.path("..", "cache")
if (!dir.exists(cache_dir)) {
  dir.create(cache_dir, recursive = TRUE)
}
# ----------
```

# Data Prep

```{r}
# Paths to data.
# --------------
dcc <- dir(
  file.path(data_dir, "external", "tau-mutant-geomx", "GeoMxPipeline"), pattern = ".dcc$", full.names = TRUE
)
pkc <- file.path(data_dir, "external", "tau-mutant-geomx", "Mm_R_NGS_WTA_v1.0.pkc")
anno <- file.path(data_dir, "external", "tau-mutant-geomx", "dufflab_20241126T1146", "AnalysisSampleSheet.xlsx")
# --------------

# Create NanoStringGeoMxSet.
# --------------------------
rds <- file.path(cache_dir, "data.rds") # Cache object as it takes a long time to create.
if (file.exists(rds)) {
  data <- readRDS(rds)
} else {
  data <- suppressWarnings(
    readNanoStringGeoMxSet(
      dcc, pkc, anno,
      phenoDataSheet = "Sheet1", protocolDataColNames = c("Genotype", "Bio_Rep", "Tech_Rep", "Slide_Rep")
    )
  )
  saveRDS(data, file = rds)
}
# --------------------------

data
```

# Quality Control

```{r}
data <- shiftCountsOne(data, useDALogic = TRUE) # Shift counts by one.

# Flag samples and probes with low sequencing depth and saturation.
# -----------------------------------------------------------------
data <- setSegmentQCFlags(data, qcCutoffs = list(percentSaturation = 45))
data <- setBioProbeQCFlags(data)

low_saturation <- which(protocolData(data)$QCFlags$LowSaturation)
colnames(assayData(data)$exprs[ , low_saturation])

low_quality_probes <- which(
  featureData(data)$QCFlags$LowProbeRatio | featureData(data)$QCFlags$GlobalGrubbsOutlier
)
length(low_quality_probes)

data <- data[ , -low_saturation] # Remove those below QC thresholds.
# -----------------------------------------------------------------

dim(data)

data <- aggregateCounts(data) # Turn probe names into gene symbols.
data.frame(assayData(data)$exprs[1:5, 1:2])

data <- normalize(data, norm_method = "quant", desiredQuantile = 0.75, toElt = "q_norm") # Normalize data.
data.frame(assayData(data)$q_norm[1:5, 1:2])
```

# Create Seurat Object

```{r}
data <- as.Seurat(data, normData = "q_norm")
data
```

# References

This is the concluding section of the document, where we output the `sessionInfo`, and create a bibliography for works cited.

```{r}
sessionInfo()
```
